plugins/fabric8-multi-tenant-manager/src/test/java/com/redhat/che/multitenant/tenantdata/TenantDataCacheLoaderTest.java

```diff
@@ -0,0 +1,158 @@
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.mockito.Mock;
+import org.mockito.testng.MockitoTestNGListener;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+
+@Listeners(MockitoTestNGListener.class)
+public class TenantDataCacheLoaderTest {
+  private static final String RESPONSE_FORMAT =
+      "{\"data\":{\"attributes\":{\"created-at\":\"createdAt\",\"namespaces\":[{\"cluster-app-domain\": %s,\"cluster-capacity-exhausted\": %b,\"cluster-url\": %s,\"name\": %s,\"type\": %s}]},\"id\":\"userid\",\"type\":\"userservices\"}}";
+  // Does not contain one of the fields (cluster-url) from above, will result in deserialization to
+  // null
+  private static final String BAD_RESPONSE_FORMAT =
+      "{\"data\":{\"attributes\":{\"created-at\":\"createdAt\",\"namespaces\":[{\"cluster-app-domain\": %s,\"cluster-capacity-exhausted\": %b,\"name\": %s,\"type\": %s}]},\"id\":\"userid\",\"type\":\"userservices\"}}";
+
+  private static final String NAMESPACE = "test-namespace";
+  private static final String ENDPOINT = "test-endpoint";
+  private static final String CLUSTER_URL = "test-url";
+  private static final String ROUTE_PREFIX = "test-route-prefix";
+  private static final String NAMESPACE_TYPE = "che";
+
+  @Mock private HttpJsonRequestFactory httpJsonRequestFactory;
+  @Mock private HttpJsonResponse httpJsonResponse;
+  @Mock private HttpJsonRequest httpJsonRequest;
+
+  private TenantDataCacheLoader cacheLoader;
+  private TenantDataCacheLoader standaloneCacheLoader;
+
+  @BeforeMethod
+  public void setUp() throws Exception {
+    when(httpJsonRequestFactory.fromUrl(any(String.class))).thenReturn(httpJsonRequest);
+    when(httpJsonRequest.setMethod(any(String.class))).thenReturn(httpJsonRequest);
+    when(httpJsonRequest.setAuthorizationHeader(any(String.class))).thenReturn(httpJsonRequest);
+    when(httpJsonRequest.request()).thenReturn(httpJsonResponse);
+
+    cacheLoader = new TenantDataCacheLoader(httpJsonRequestFactory, ENDPOINT, NAMESPACE, false);
+    standaloneCacheLoader =
+        new TenantDataCacheLoader(httpJsonRequestFactory, ENDPOINT, NAMESPACE, true);
+  }
+
+  @Test
+  public void shouldParseJsonAndReturnTenantData() throws Exception {
+    // Given
+    String jsonResponse =
+        generateResponse(ROUTE_PREFIX, CLUSTER_URL, NAMESPACE, NAMESPACE_TYPE, false);
+    when(httpJsonResponse.asString()).thenReturn(jsonResponse);
+    TenantDataCacheKey cacheKey = new TenantDataCacheKey("token", "che");
+
+    // When
+    UserCheTenantData data = cacheLoader.load(cacheKey);
+
+    // Then
+    assertEquals(data.getRouteBaseSuffix(), ROUTE_PREFIX);
+    assertEquals(data.getClusterUrl(), CLUSTER_URL);
+    assertEquals(data.getNamespace(), NAMESPACE);
+    assertEquals(data.isClusterCapacityExhausted(), false);
+  }
+
+  @Test(expectedExceptions = InfrastructureException.class)
```

Comment:
From my experience tests that expect exception but do not specify exception message tend to degrade after new changes because some another error is caught instead of the error that was supposed to be caught by the test. Consider adding exception message.

Reply:
Added regexes.